azerothcore/azerothcore-wotlk
# feat(Core/WorldSession): implement mute manager

## Changes Proposed:
- Added config option `Mute.AddAfterLogin` for mute now or after login if the target offline
- Removed old mute system (from `account` db table)
- Added new functions for time format

## Issues Addressed:
- Closes 

## Tests Performed:
![image](https://user-images.githubusercontent.com/18329778/132942229-e991800f-8dd9-4de7-82aa-4429d9248fa7.png)
![image](https://user-images.githubusercontent.com/18329778/132956854-f22de8c7-20e1-4b27-9323-28abea59514b.png)

## How to Test the Changes:
1. Try add mute any player
2. Check speak

## Known Issues and TODO List:
- [x] to AC
- [x] change log
- [x] columns
- [x] https://github.com/azerothcore/mod-ah-bot/pull/40
- [x] offline mute
- [ ] `After removing the old mute system, will the original muted player be muted in the new system?`
- [ ] Wait merge #8630

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
